File: files/400-499/436/CH-436.py
import pandas as pd

path = "400-499/436/CH-436 Number Series.xlsx"
input = pd.read_excel(path, usecols="B", nrows=8, skiprows=2)
test = pd.read_excel(path, usecols="D", nrows=8, skiprows=2)


def pell_index(n):
    a, b, i = 0, 1, 0
    while a < n:
        a, b, i = b, 2 * b + a, i + 1
    return float(i) if a == n else float("nan")


result = input.copy()
result["n"] = result["Pell Number"].apply(pell_index)

print(result["n"].equals(test["n"]))
# True

File: files/400-499/436/CH-436.R
library(tidyverse)
library(readxl)

path <- "400-499/436/CH-436 Number Series.xlsx"
input <- read_excel(path, range = "B3:B10")
test <- read_excel(path, range = "D3:D10", col_types = "numeric")

pell_index <- function(n) {
  p <- c(0, 1)
  while (tail(p, 1) < n) {
    p <- c(p, 2 * p[length(p)] + p[length(p) - 1])
  }
  i <- which(p == n) - 1
  if (length(i)) as.numeric(i) else NA_real_
}

result <- input %>%
  mutate(n = map_dbl(`Pell Number`, pell_index))

all.equal(result$n, test$n)
# True
